src/js/_enqueues/lib/nav-menu.js

```diff
@@ -1103,12 +1103,38 @@
 
 			$('#add-custom-links input[type="text"]').on( 'keypress', function(e){
 				$('#customlinkdiv').removeClass('form-invalid');
+				$('#custom-menu-item-url').hide();
 
 				if ( e.keyCode === 13 ) {
 					e.preventDefault();
 					$( '#submit-customlinkdiv' ).trigger( 'click' );
 				}
 			});
+
+			$('#submit-customlinkdiv').on('click', function (e) {
+				var urlInput = $('#custom-menu-item-url'),
+					url = urlInput.val().trim(),
+					errorMessage = $('#custom-url-error'),
+					urlWrap = $('#menu-item-url-wrap');
+
+				// Hide the error message initially
+				errorMessage.hide();
+				urlWrap.removeClass('has-error');
+
+				if ('' === url || 'https://' === url || 'http://' === url) {
+					e.preventDefault();
+					urlInput.addClass('form-invalid')
+						.attr('aria-invalid', 'true')
+						.attr('aria-describedby', 'custom-url-error');
+
+					var errorText = wp.i18n.__('No content entered. Please enter a valid link.');
+					errorMessage.show();
```

Comment:
I know I suggested this text, but given the flexibility in what the possible values can be, probably need to make this more general yet. Perhaps "Content is required for the link's `href` attribute."